### PR TITLE
Send over namespace

### PR DESCRIPTION
**Pull request by danielmarbach**
Describes the new API introduced in https://github.com/Particular/NServiceBus.AzureServiceBus/pull/623

The package hasn't been released so this build is expected to fail.

### CODE REVIEW

File: Snippets/ASB/ASB_8/MultipleNamespaces.cs [hunk 1]
```diff
@@ -80,6 +80,24 @@ void NamespaceRoutingRegistration(EndpointConfiguration endpointConfiguration)
         #endregion
     }
 
+    void NamespaceRoutingEndpointRegistration(EndpointConfiguration endpointConfiguration)
+    {
+        #region namespace_routing_endpoint_registration
+
+        var transport = endpointConfiguration.UseTransport<AzureServiceBusTransport>();
+
+        var routing = transport.Routing();
```
Comment (SeanFeldman) [inline Snippets/ASB/ASB_8/MultipleNamespaces.cs]: I'd add comments with steps. Something among the lines of 
// Step 1 - associate MyMessage with "sales" endpoint
...
// Step 2 - register namespace that will be used for "sales" endpoint
...
// Step 3 - Associated endpoint "sales" with the namespace to route MyMessage messages to

File: transports/azure-service-bus/multiple-namespaces-support_registered-endpoint_asb_[8,).partial.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Versions 8 and above of the Azure Service Bus transport a more convenient way of sending messages over multiple namespaces has been introduced that does not required to explicitely specify the destination.
```
Comment (SeanFeldman) [inline transports/azure-service-bus/multiple-namespaces-support_registered-endpoint_asb_[8,).partial.md]: > Versions 8 and above of the Azure Service Bus transport a more convenient way

"Offer" or "provide" is missing

File: transports/azure-service-bus/multiple-namespaces-support_registered-endpoint_asb_[8,).partial.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+To leverage that capability the namespace alias and connection string needs to be registered using the `NamespaceRouting()` API. Furthermore the endpoint name needs to be registered on the namespace that it belongs to.
```
Comment (SeanFeldman) [inline transports/azure-service-bus/multiple-namespaces-support_registered-endpoint_asb_[8,).partial.md]: need~~s~~

Comment (SeanFeldman) [reply transports/azure-service-bus/multiple-namespaces-support_registered-endpoint_asb_[8,).partial.md]: Furthermore **,**

**Comment by weralabaj** [comment]
> The package hasn't been released so this build is expected to fail.

Can you mark this as a WIP then? Thx.

**Comment by danielmarbach** [comment]
@SeanFeldman addressed your comments

**Comment by SeanFeldman** [comment]
Do we need a new beta on myget for the doco to work @danielmarbach ?

**Comment by SeanFeldman** [comment]
Will push beta-0002 out and merge this one.

**Comment by SeanFeldman** [comment]
Merged.

**Pull request merged**